streaming/src/test/scala/org/apache/spark/streaming/FailureSuite.scala

```diff
@@ -43,6 +43,9 @@ class FailureSuite extends SparkFunSuite with BeforeAndAfter with Logging {
       Utils.deleteRecursively(directory)
     }
     StreamingContext.getActive().foreach { _.stop() }
+
+    // Stop SparkContext if active
+    SparkContext.getOrCreate(new SparkConf().setMaster("local").setAppName("bla")).stop()
```

Comment:
I am not entirely sure. I think somehow a StreamingContext is not shutdown cleanly, which causes the SparkContext to leak. This fix at least prevents the issue to cascade to other testsuites.